### PR TITLE
WIP: Decouple TR_PrexArgInfo from TR_InlinerTracer

### PR DESCRIPTION
**Pull request by rpshukla**
Move code from TR_InlinerTracer into TR_PrexArgInfo class so that TR_PrexArgInfo is no longer tightly couple to the TR_InlinerTracer class.

This should make the TR_PrexArgInfo class more reusable.

Note: this should be merged at the same time as https://github.com/eclipse/omr/pull/4779

- Replace calls to `TR_InlinerTracer::dumpPrexArgInfo` with calls to `TR_PrexArgInfo::dumpTrace`
- Access comp through static method `TR::comp` instead of using `tracer->comp()`
- Modify `TR_InterpreterEmulator` class to use the `TR_LogTracer` class instead of `TR_InlinerTracer` since `TR_LogTracer` is a less specialized class

Fixes: #7936

Signed-off-by: Ryan Shukla <ryans@ibm.com>

### CODE REVIEW

**Comment by rpshukla** [comment]
@cathyzhyi

File: runtime/compiler/optimizer/InlinerTempForJ9.cpp [hunk 1]
```diff
@@ -5103,11 +5103,11 @@ void TR_J9InlinerUtil::checkForConstClass(TR_CallTarget *target, TR_InlinerTrace
    TR_PrexArgInfo *ecsArgInfo = target->_ecsPrexArgInfo;
    if (!ecsArgInfo) return;
 
-   TR::Compilation * comp = tracer->comp();
+   TR::Compilation * comp = TR::comp();
    bool tracePrex = comp->trace(OMR::inlining) || comp->trace(OMR::invariantArgumentPreexistence);
 
    if (tracePrex)
-      traceMsg(comp, "checkForConstClass parm for [%p] %s %s\n", callNode, callNode->getOpCode().getName(), callNode->getSymbol()->castToMethodSymbol()->getMethod()->signature(tracer->trMemory(), stackAlloc));
+      traceMsg(comp, "checkForConstClass parm for [%p] %s %s\n", callNode, callNode->getOpCode().getName(), callNode->getSymbol()->castToMethodSymbol()->getMethod()->signature(TR::comp()->trMemory(), stackAlloc));
```
Comment (cathyzhyi) [inline runtime/compiler/optimizer/InlinerTempForJ9.cpp]: It's better to use the local variable `comp` when it's already available. `TR::comp()` is in special thread-local memory and calling it each time can add overhead.

File: runtime/compiler/optimizer/InlinerTempForJ9.cpp [hunk 2]
```diff
@@ -5187,14 +5187,14 @@ void TR_J9InlinerUtil::checkForConstClass(TR_CallTarget *target, TR_InlinerTrace
...
-TR_PrexArgInfo* TR_PrexArgInfo::buildPrexArgInfoForMethodSymbol(TR::ResolvedMethodSymbol* methodSymbol, TR_InlinerTracer* tracer)
+TR_PrexArgInfo* TR_PrexArgInfo::buildPrexArgInfoForMethodSymbol(TR::ResolvedMethodSymbol* methodSymbol, TR_LogTracer* tracer)
    {
    int numArgs = methodSymbol->getParameterList().getSize();
    TR_ResolvedMethod       *feMethod = methodSymbol->getResolvedMethod();
    ListIterator<TR::ParameterSymbol> parms(&methodSymbol->getParameterList());
 
-   TR_PrexArgInfo *argInfo = new (tracer->trHeapMemory()) TR_PrexArgInfo(numArgs, tracer->trMemory());
-   heuristicTrace(tracer, "PREX-CSI:  Populating parmInfo of current method %s\n", feMethod->signature(tracer->trMemory()));
+   TR_PrexArgInfo *argInfo = new (TR::comp()->trHeapMemory()) TR_PrexArgInfo(numArgs, TR::comp()->trMemory());
```
Comment (cathyzhyi) [inline runtime/compiler/optimizer/InlinerTempForJ9.cpp]: actually we can replace calling `TR::comp()` here with `tracer->comp()` if we have to keep `TR_LogTracer` anyways.

**Comment by cathyzhyi** [comment]
All the changed files needs to change the copy right line to 2020 like this https://github.com/eclipse/openj9/blob/c09da40ba632048d383fe2525db2296ac95ae5c6/runtime/compiler/optimizer/InlinerTempForJ9.cpp#L2

**Comment by cathyzhyi** [comment]
Also, it's preferable to squash the commits into one commit in a single PR. Especially, in this case, the commits are fairly straight forward.

**Comment by rpshukla** [comment]
Update: I am going to close this PR and open a new one to stage the changes as described here: https://github.com/eclipse/omr/pull/4779#issuecomment-594893313

This will eliminate the need to merge simultaneously with omr.

**Pull request closed without merging**